src/views/Notifications/Settings/index.tsx

```diff
@@ -85,6 +96,18 @@ const NotificationSettings = () => {
       ...dirtyInputs
     };
 
+    // if datesChangedNotificationType is not changed by user, but datesChanged is changed, then do the following logic
+    if (!changes.datesChangedNotificationType) {
+      if (changes.datesChanged?.length) {
+        // If Dates Changed notification is subscribed, then manually set datesChangedNotificationType to ALL_MODELS
+        changes.datesChangedNotificationType =
+          DatesChangedNotificationType.ALL_MODELS;
+      } else {
+        // If Dates Changed notification is unsubscribed, set datesChangedNotificationType to null
```

Comment:
I think this is part of the culprit.  When the page is first hit by the email link, `datesChangedNotificationType` is not null, the mutation fires and it get set to null.  Then once rechecked, it appears to not be dirty anymore, hence it explicitly sets it to null